src/utils/components/SSHAccess/SSHAccessModal.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import { Trans } from 'react-i18next';
+
+import { ServiceModel } from '@kubevirt-ui/kubevirt-api/console';
+import VirtualMachineModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineModel';
+import {
+  IoK8sApiCoreV1Service,
+  IoK8sApiCoreV1ServiceSpecTypeEnum,
+} from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { k8sCreate, k8sDelete, ResourceLink } from '@openshift-console/dynamic-plugin-sdk';
+import { Alert, Checkbox, ModalVariant, Stack, StackItem } from '@patternfly/react-core';
+
+import ExternalLink from '../ExternalLink/ExternalLink';
+import TabModal from '../TabModal/TabModal';
+
+type SSHAccessModalProps = {
+  vm: V1VirtualMachine;
+  isOpen: boolean;
+  onClose: () => void;
+  sshService: IoK8sApiCoreV1Service;
+};
+
+export const NODE_PORTS_LINK =
+  'https://access.redhat.com/documentation/en-us/openshift_container_platform/4.9/html/networking/configuring-ingress-cluster-traffic#nw-using-nodeport_configuring-ingress-cluster-traffic-nodeport';
+
+export const PORT = 22000;
+export const TARGET_PORT = 22;
+
+const getSSHServiceFromVM = (vm: V1VirtualMachine): IoK8sApiCoreV1Service => ({
+  kind: ServiceModel.kind,
+  apiVersion: ServiceModel.apiVersion,
+  metadata: {
+    name: `${vm?.metadata?.name}-ssh-service`,
+    namespace: vm?.metadata?.namespace,
+  },
+  spec: {
+    ports: [
+      {
+        port: PORT,
+        targetPort: TARGET_PORT.toString(),
+      },
+    ],
+    type: IoK8sApiCoreV1ServiceSpecTypeEnum.NodePort,
+    selector: {
+      ...Object.fromEntries(
+        Object.entries(vm?.metadata?.labels).filter(
+          ([objectKey]) => !objectKey.startsWith('vm') && !objectKey.startsWith('app'),
+        ),
+      ),
+      'kubevirt.io/domain': vm?.metadata?.name,
+      'vm.kubevirt.io/name': vm?.metadata?.name,
+    },
+  },
+});
+
+const SSHAccessModal: React.FC<SSHAccessModalProps> = ({ vm, isOpen, onClose, sshService }) => {
+  const { t } = useKubevirtTranslation();
+  const [isEnabled, setEnabled] = React.useState<boolean>(!!sshService);
+
+  const onSubmit = async () => {
+    const serviceResource = getSSHServiceFromVM(vm);
+
+    try {
+      if (isEnabled) {
+        await k8sCreate<IoK8sApiCoreV1Service>({
+          model: ServiceModel,
+          data: serviceResource,
+          ns: vm?.metadata?.namespace,
+        });
+      } else {
+        await k8sDelete({
+          model: ServiceModel,
+          resource: serviceResource,
+          json: undefined,
+          requestInit: undefined,
+        });
+      }
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.log(`Failed to ${isEnabled ? 'create' : 'delete'} SSH service:`, e.message);
+    }
+  };
```

Comment:
im guessing that in a scenrio when a user just open modal and click onsubmit without changing anything there will be still unneccessry calls, I think we should harden the conditions for the call submits, such as isEnabled && !!sshService , and same goes for delete, !isEnabled && !sshService